src/tiled/mapview.cpp

```diff
@@ -214,8 +214,9 @@ void MapView::wheelEvent(QWheelEvent *event)
     auto *hBar = static_cast<FlexibleScrollBar*>(horizontalScrollBar());
     auto *vBar = static_cast<FlexibleScrollBar*>(verticalScrollBar());
 
-    if (event->modifiers() & Qt::ControlModifier
-        && event->orientation() == Qt::Vertical)
+    if ((static_cast<bool>(event->modifiers() & Qt::ControlModifier) ^
+         static_cast<bool>(Preferences::instance()->zoomWheelByDefault()))
```

Comment:
For readability sake, please introduce local variables like `control` and `wheelZoomsByDefault`, so you can do:
```c++
if ((wheelZoomsByDefault != control) && event->orientation() == Qt::Vertical) {
```